sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/filter/AadAppRoleStatelessAuthenticationFilter.java

```diff
@@ -97,8 +97,7 @@ protected void doFilterInternal(HttpServletRequest httpServletRequest,
             // Invalid JWT. Either expired or not yet valid.
             httpServletResponse.sendError(HttpStatus.UNAUTHORIZED.value());
         } catch (ParseException | BadJOSEException | JOSEException ex) {
-            LOGGER.error("Failed to initialize UserPrincipal.", ex);
-            throw new ServletException(ex);
```

Comment:
Fix issue: Either log this exception and handle it, or rethrow it with some contextual information.